spec/lib/tasks/provider_transfer/ptr_migrations_spec.rb

```diff
@@ -0,0 +1,179 @@
+    end
+
+    context "when LegalAidApplication and Office exist" do
+      it "transfers the LegalAidApplication to the new office" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication (#{legal_aid_application.application_ref}) current office code: #{initial_office.code}")
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication (#{legal_aid_application.application_ref}) successfully transferred to office code: #{new_office.code}")
+
+        task.execute(mock: "false", appid: legal_aid_application.id, officeid: new_office.id)
+
+        expect(legal_aid_application.reload.office).to eq(new_office)
+      end
+    end
+
+    context "when LegalAidApplication does not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication or OFFICE does not exists")
+
+        task.execute(mock: "false", appid: nil, officeid: new_office.id)
+      end
+    end
+
+    context "when Office does not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication or OFFICE does not exists")
+
+        task.execute(mock: "false", appid: legal_aid_application.id, officeid: nil)
+      end
+    end
+
+    context "when both LegalAidApplication and Office do not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication or OFFICE does not exists")
+
+        task.execute(mock: "false", appid: nil, officeid: nil)
+      end
+    end
+
+    context "when mock is true" do
+      it "logs mock transfer messages without updating the records" do
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication (#{legal_aid_application.application_ref}) current office code: #{initial_office.code}")
+        expect(Rails.logger).to receive(:info).with("LegalAidApplication (#{legal_aid_application.application_ref}) successfully transferred to office code: #{initial_office.code}")
+
+        task.execute(mock: "true", appid: legal_aid_application.id, officeid: new_office.id)
+
+        expect(legal_aid_application.reload.office).to eq(initial_office)
+      end
+    end
+  end
+
+  describe "ptr_migrations:provider_transfer_firm" do
+    subject(:task) do
+      Rake::Task["ptr_migrations:provider_transfer_firm"]
+    end
+
+    context "when Provider and Firm exist" do
+      it "transfers the Provider to the new Firm" do
+        expect(Rails.logger).to receive(:info).with("Provider (#{provider.email}) current firm #{initial_firm.name}")
+        expect(Rails.logger).to receive(:info).with("Provider (#{provider.email}) successfully transferred to firm #{new_firm.name}")
+
+        task.execute(mock: "false", providerid: provider.id, firmid: new_firm.id)
+
+        expect(provider.reload.firm).to eq(new_firm)
+        expect(provider.reload.selected_office_id).to be_nil
+      end
+    end
+
+    context "when Provider does not exist" do
```

Comment:
Is it worth adding an `expect` to verify that provider doesn't change in these scenarios?

Reply:
ditto